x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/helper.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, pick, isEmpty } from 'lodash';
+import { Section as SectionType } from './sections';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { APMError } from '../../../../typings/es_schemas/ui/APMError';
+import { Span } from '../../../../typings/es_schemas/ui/Span';
+import { flattenObject, KeyValuePair } from '../../../utils/flattenObject';
+
+export type MetadataItems = Array<SectionType & { rows?: KeyValuePair[] }>;
+
+export const getMetadataItems = (
+  sections: SectionType[],
+  apmDoc: Transaction | APMError | Span
+): MetadataItems => {
+  return sections
+    .map(section => {
+      const sectionData: Record<string, unknown> = get(apmDoc, section.key);
+      const sectionProperties:
```

Comment:
This is not the "sectionProperties" is it? It's still "sectionData" just potentially filtered by certain properties, right?